carbon-design-system/carbon
# chore(tile-group): refactor to functional component

Ref #9712

Refactor the TileGroup component to the functional component using `React.forwardRef` to ensure refs were being forwarded appropriately.

- [x] TileGroup

#### Notes
`getDerivedStateFromProps` is modified using the recommendation in #9712. `renderLegend` is moved into the rendering method.

#### Testing / Reviewing

- Pull down the PR locally
- Run `yarn start:v11` from `packages/react`
- Alternate: use the PR storybook: https://deploy-preview-9955--carbon-components-react.netlify.app/ and v11 next storybook: https://deploy-preview-9955--carbon-react-next.netlify.app/
- Ensure that the following stories for Tile are working as expected: 
   - Radio
   - RadioWithLayer

#### Todo

- [ ] re-write tests to use RTL
